### PR TITLE
Task s2 display categories

### PR DESCRIPTION
**Pull request by karimkefi**
create class of Categories, display in Index.php and test of classes

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -1,4 +1,12 @@
-<?php require_once __DIR__ . '/vendor/autoload.php'; ?>
+<?php require_once __DIR__ . '/vendor/autoload.php';
```
Comment (JamesCullen2) [inline index.php]: could we move this onto a different line to the opening <? php tag?

File: src/classes/DBConnect.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace store;
+
+use \PDO;
+use \Exception;
+
+class DBConnect {
+    /**
+     *Connects to DB using PDO object.
+     */
```
Comment (JamesCullen2) [inline src/classes/DBConnect.php]: do we need a return in this doc block?

File: index.php [hunk 3]
```diff
@@ -1,4 +1,12 @@
...
+use Store\Category as Category;
+use Store\DBConnect as DBConnect;
+use Store\Store as Store;
+$dBConnect = DBConnect::connectToDB();
+$store = new Store($dBConnect);
```
Comment (JamesCullen2) [inline index.php]: could we introduce a static function :: to link new Store with getCategories method?

File: src/classes/Store.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        $query = $this->dBConnection->prepare("SELECT `id`, `categoryName`, `defaultImageFilePath`,`defaultImageAlt` FROM `categories` WHERE `deleted` = 0;");
+        $query->execute();
+        return $query->fetchAll(PDO::FETCH_CLASS, Category::class);
+    }
+
+    /**
+     * Store constructor, adds db connection to object.
+     * @param $db PDO object
+     */
+    public function __construct(PDO $db) {
```
Comment (JamesCullen2) [inline src/classes/Store.php]: where does $db variable come from / how is it passed in here ? (this is more a general learning question than a critique of code !)

File: src/classes/Category.php [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+    /**
+     * Generates html for category list.
+     *
+     * @return string
+     */
+    public function getCategoryListLink():string {
+        return "<a href='" .
+            $this->path .
+            $this->id.
```
Comment (JamesCullen2) [inline src/classes/Category.php]: No space after id before .

File: src/classes/Category.php [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+     * Generates html for category blocks.
+     *
+     * @return string
+     */
+    public function getCategoryImageLink():string {
+        return "<div class=\"col-xs-8 col-sm-4\">
+                <a href='" .
+                $this->path .
+                $this->id .
+                "'><img class='img-thumbnail' src='".
```
Comment (JamesCullen2) [inline src/classes/Category.php]: No space after src"" before .

File: src/classes/Category.php [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+                $this->defaultImageFilePath.
```
Comment (JamesCullen2) [inline src/classes/Category.php]: No space after defaultImageFilePath before .

**Review by JamesCullen2 — CHANGES_REQUESTED** [review]
Looks really good clean code, well done gents !

File: index.php (hunk 1 above)
Comment (karimkefi) [reply index.php]: done

File: src/classes/DBConnect.php (hunk 2 above)
Comment (karimkefi) [reply src/classes/DBConnect.php]: done

File: index.php (hunk 3 above)
Comment (karimkefi) [reply index.php]: done

File: src/classes/Store.php (hunk 4 above)
Comment (karimkefi) [reply src/classes/Store.php]: removed due to static function implementation

File: src/classes/Category.php (hunk 6 above)
Comment (karimkefi) [reply src/classes/Category.php]: done for all .

File: src/classes/Category.php (hunk 5 above)
Comment (karimkefi) [reply src/classes/Category.php]: done for all .

File: src/classes/Category.php (hunk 7 above)
Comment (karimkefi) [reply src/classes/Category.php]: done for all .

**Review by JamesCullen2 — APPROVED** [review]

**Pull request merged**